src/averageAges.js

```diff
@@ -15,11 +16,20 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const speacialMen = people.filter(person =>
+    Math.ceil(person.died / 100) === century && person.sex === 'm');
+
+  const speacialSum = speacialMen.reduce((sum, { died, born }) =>
+    (sum + (died - born)), 0);
+
+  const allMen = people.filter(person => person.sex === 'm');
+
+  const generalSum = allMen.reduce((sum, { died, born }) =>
+    (sum + (died - born)), 0);
```

Comment:
it would be better to check if there is `century` argument in the very beginning and make all calculations according to it, to avoid unnecessary actions